tile-generation/src/main/scala/com/oculusinfo/tilegen/tiling/TileOperations.scala

```diff
@@ -0,0 +1,353 @@
+    tilePipeline.registerPipelineOp("cache", parseCacheDataOp)
+    tilePipeline.registerPipelineOp("date_filter", parseDateFilterOp)
+    tilePipeline.registerPipelineOp("integral_range_filter", parseIntegralRangeFilterOp)
+    tilePipeline.registerPipelineOp("fractional_range_filter", parseFractionalRangeFilterOp)
+    tilePipeline.registerPipelineOp("regex_filter", parseRegexFilterOp)
+  }
+
+  /**
+   * Parse data path from args
+   */
+  def parseLoadJsonDataOp(args: Map[String, String]) = {
+    val argParser = KeyValuePassthrough(args)
+    val path = argParser.getString("ops.path", "HDFS path to data")
+    loadJsonDataOp(path)_
+  }
+
+  /**
+   * Load data from a JSON file.  The schema is derived from the first json record.
+   */
+  def loadJsonDataOp(path: String)(data: PipelineData): PipelineData = {
+    PipelineData(data.sqlContext, data.sqlContext.jsonFile(path))
+  }
+
+  /**
+   * Parse data path from args
+   */
+  def parseLoadCsvDataOp(args: Map[String, String]) = {
+    val argParser = KeyValuePassthrough(args)
+    val path = argParser.getString("ops.path", "HDFS path to data")
+    loadCsvDataOp(path, argParser)_
+  }
+
+  /**
+   * Load data from a CSV file.  The schema is derived from the first json record.
+   */
+  def loadCsvDataOp(path: String, argumentSource: KeyValueArgumentSource)(data: PipelineData): PipelineData = {
+    val reader = new CSVReader(data.sqlContext, path, argumentSource)
+    PipelineData(reader.sqlc, reader.asSchemaRDD)
+  }
+
+  /**
+   * Parse start / end times from args.
+   */
+  def parseDateFilterOp(args: Map[String, String]) = {
+    val argParser = KeyValuePassthrough(args)
+    val timeCol = argParser.getString("ops.column", "Col spec denoting the time field in input data")
+    val start = argParser.getString("ops.start", "Start date")
+    val end = argParser.getString("ops.end", "End date")
+    val formatter = argParser.getString("ops.format", "Date format string")
+    dateFilterOp(start, end, formatter, timeCol)_
+  }
+
+  /**
+   * Pipeline op to filter records to a specific date range
+   */
+  def dateFilterOp(minDate: String, maxDate: String, format: String, timeCol: String)(input: PipelineData) = {
+    val formatter = new SimpleDateFormat(format)
+    val minTime = formatter.parse(minDate).getTime
+    val maxTime = formatter.parse(maxDate).getTime
+    val timeExtractor = calculateExtractor(timeCol, input.srdd.schema)
+    val filtered = input.srdd.filter { row =>
+      val time = formatter.parse(timeExtractor(row).toString).getTime
+      minTime <= time && time <= maxTime
+    }
+    PipelineData(input.sqlContext, filtered)
+  }
+
+  /**
+   * Parses args for cache data.  No arguments required.
+   */
+  def parseCacheDataOp(args: Map[String, String]) = {
+    cacheDataOp()_
+  }
+
+  /**
+   * Pipeline op to cache data
+   */
+  def cacheDataOp()(input: PipelineData) = {
+    PipelineData(input.sqlContext, input.srdd.cache())
```

Comment:
While I think this is fixed in some version of spark, in 1.1.1 at least, I believe we need to register a table and call sqlContext.cacheTable(tableName) to actually get it to cache.